mux/listen.go

```diff
@@ -9,7 +9,7 @@ type Listener interface {
 	Close() error
 
 	// Accept waits for and returns the next incoming session.
-	Accept() (*Session, error)
+	Accept() (Session, error)
```

Comment:
One thing I noticed is that the quic implementation has Accept() and AcceptStream() take a Context to enable cancelling. So we may want to consider whether that is helpful here before finalizing the interfaces.

Reply:
Yea that's usually the way to go. It's kind of a rare case here. Usually you set something up to Accept and leave it until you Close or get an EOF. If I can avoid a context, I usually do but what do you think.

Reply:
Yeah I agree it doesn't seem super common for Accept. Looks like the discussion on the QUIC library goes back to lucas-clemente/quic-go#1915 where he went ahead and added Context to each blocking call.